src/clients/c/tb_client/signal_fuzz.zig

```diff
@@ -0,0 +1,135 @@
+const std = @import("std");
+const assert = std.debug.assert;
+const log = std.log.scoped(.fuzz_signal);
+
+const Signal = @import("./signal.zig").Signal;
+const IO = @import("../../../io.zig").IO;
+const stdx = @import("../../../stdx.zig");
+const fuzz = @import("../../../testing/fuzz.zig");
+
+const Context = struct {
+    const Atomic = std.atomic.Value(enum(u8) {
+        none,
+        from_io_thread,
+        from_user_thread,
+    });
+
+    io: IO,
+    main_thread_id: std.Thread.Id,
+    signal: Signal,
+
+    random: std.Random,
+    running_count: u32 = 0,
+    stop_request: Atomic = Atomic.init(.none),
+};
+
+const Threads = stdx.BoundedArrayType(std.Thread, threads_limit);
+
+const threads_limit = 8;
+const stop_chance_percentage = 1;
+
+pub fn main(args: fuzz.FuzzArgs) !void {
+    var prng = std.rand.DefaultPrng.init(args.seed);
+    const events_max = args.events_max orelse 10_000;
+
+    var context: Context = .{
+        .io = try IO.init(32, 0),
+        .main_thread_id = std.Thread.getCurrentId(),
+        .signal = undefined,
+        .random = prng.random(),
+    };
+    defer context.io.deinit();
```

Comment:
This should be a relatively fast fuzzer, as it doesn't make much sense to do more than a dozen events within a single run. So let's add an overall-loop here:

```
for (0..iterations_total) {
    const io = try IO.init(32, 0)
    for (0..notifies_count) {
    }
}
```

That way, we'll test more `cancel` calls!